pkg/diff/util.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diff
+
+import (
+	"math/rand"
+
+	"github.com/ngaut/log"
+	"github.com/pingcap/tidb-tools/pkg/dbutil"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/types"
+)
+
+func equalStrings(str1, str2 []string) bool {
+	if len(str1) != len(str2) {
+		return false
+	}
+	for i := 0; i < len(str1); i++ {
+		if str1[i] != str2[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func removeColumns(tableInfo *model.TableInfo, columns []string) *model.TableInfo {
+	if len(columns) == 0 {
+		return tableInfo
+	}
+	removeColMap := SliceToMap(columns)
+	for i, index := range tableInfo.Indices {
+		for j, col := range index.Columns {
+			if _, ok := removeColMap[col.Name.O]; ok {
+				index.Columns = append(index.Columns[:j], index.Columns[j+1:]...)
+				if len(index.Columns) == 0 {
+					tableInfo.Indices = append(tableInfo.Indices[:i], tableInfo.Indices[i+1:]...)
+				}
+			}
+		}
+	}
+
+	for j, col := range tableInfo.Columns {
+		if _, ok := removeColMap[col.Name.O]; ok {
+			tableInfo.Columns = append(tableInfo.Columns[:j], tableInfo.Columns[j+1:]...)
+		}
+	}
+
+	return tableInfo
+}
+
+func getRandomN(total, num int) []int {
+	if num > total {
+		log.Warnf("the num %d is greater than total %d", num, total)
+		num = total
+	}
+
+	totalArray := make([]int, 0, total)
+	for i := 0; i < total; i++ {
+		totalArray = append(totalArray, i)
+	}
+
+	for j := 0; j < num; j++ {
+		r := j + rand.Intn(total-j)
+		totalArray[j], totalArray[r] = totalArray[r], totalArray[j]
+	}
+
+	return totalArray[:num]
+}
+
+func needQuotes(ft types.FieldType) bool {
```

Comment:
number & float can use quota too

Reply:
I will remove this function

Reply:
but "2"> "11" will return true, I will keep this function